perfkitbenchmarker/linux_benchmarks/cluster_boot_benchmark.py

```diff
@@ -58,7 +58,7 @@ def _GetTimeToBoot(vms, vm_index):
   """
   vm = vms[vm_index]
   metadata = {'num_cpus': vm.num_cpus, 'machine_instance': vm_index,
-              'num_vms': len(vms)}
+              'num_vms': len(vms), 'os_type': vm.OS_TYPE}
```

Comment:
I don't think it requires that the VMs be homogenous.  A sample is added per-vm, so one cluster boot benchmark with 20 vms generates 21 samples--one for the boot time for each vm, and one for the the end-to-end runtime.     The os_type metadata is only added to the vm-specific record, and not to the end-to-end runtime record.